pooch/processors.py

```diff
@@ -50,36 +52,113 @@ def __call__(self, fname, action, pooch):
+
+
+    Parameters
+    ----------
+    members : list or None
+        If None, will unpack all files in the zip archive. Otherwise, *members* must be
+        a list of file names to unpack from the archive. Only these files will be
+        unpacked.
+    """
+
+    suffix = ".unzip"
+
+    def _extract_file(self, fname, extract_dir):
+        """
+        This method receives an argument for the archive to extract and the
+        destination path.
+        """
+        with ZipFile(fname, "r") as zip_file:
+            if self.members is None:
+                warn("Unzipping contents of '{}' to '{}'".format(fname, extract_dir))
+                # Unpack all files from the archive into our new folder
+                zip_file.extractall(path=extract_dir)
+            else:
+                for member in self.members:
+                    warn(
+                        "Extracting '{}' from '{}' to '{}'".format(
+                            member, fname, extract_dir
+                        )
+                    )
+                    # Extract the data file from within the archive
+                    with zip_file.open(member) as data_file:
+                        # Save it to our desired file name
+                        with open(os.path.join(extract_dir, member), "wb") as output:
+                            output.write(data_file.read())
+
+
+class Untar(ExtractorProcessor):  # pylint: disable=too-few-public-methods
+    """
+    Processor that unpacks a tar archive and returns a list of all files.
+
+    Use with :meth:`pooch.Pooch.fetch` to untar a downloaded data file into a folder in
+    the local data store. :meth:`~pooch.Pooch.fetch` will return a list with the names
+    of the extracted files instead of the archive.
+
+    The output folder is ``{fname}.untar``.
+
+
+    Parameters
+    ----------
+    members : list or None
+        If None, will unpack all files in the archive. Otherwise, *members* must be
+        a list of file names to unpack from the archive. Only these files will be
+        unpacked.
+    """
+
+    suffix = ".untar"
+
+    def _extract_file(self, fname, extract_dir):
+        """
+        This method receives an argument for the archive to extract and the
+        destination path.
+        """
+        with TarFile.open(fname, "r") as tar_file:
+            if self.members is None:
+                warn("Untarring contents of '{}' to '{}'".format(fname, extract_dir))
+                # Unpack all files from the archive into our new folder
+                tar_file.extractall(path=extract_dir)
+            else:
+                for member in self.members:
+                    warn(
+                        "Extracting '{}' from '{}' to '{}'".format(
+                            member, fname, extract_dir
+                        )
+                    )
+                    # Extract the data file from within the archive
+                    with tar_file.getmember(member) as data_file:
```

Comment:
```suggestion
                    with tar_file.extractfile(member) as data_file:
```

I think `getmember` only returns metadata, not the actual contents. `extractfile` returns the member as a file-like object that we can `.read()` later.

Reply:
Ah!  OK, thank you -- and this makes it easier to then call `read`, as suggested below.